clients/drcachesim/tools/basic_counts.cpp

```diff
@@ -294,3 +293,38 @@ basic_counts_t::get_total_counts()
     }
     return total;
 }
+
+bool
+basic_counts_t::parallel_shard_quantum_end(void *shard_data, uint64_t quantum_id)
+{
+    per_shard_t *per_shard = reinterpret_cast<per_shard_t *>(shard_data);
+    counters_t shard_total;
+    for (const auto &ctr : per_shard->counters) {
+        shard_total += ctr;
+    }
+    std::cerr << "Saw "
+              << shard_total.instrs - per_shard->last_quantum_counters_snapshot.instrs
+              << " instrs in quantum " << quantum_id << " of TID " << per_shard->tid
+              << "\n";
+    per_shard->last_quantum_counters_snapshot = shard_total;
```

Comment:
This is a little confusing: it's summing up all the counts from all prior windows and time series snapshots, but then it just prints out the relative change in instructions.  Are these snapshots supposed to be relative, or absolute?  Is there a convention to be documented?  What about the other counters besides instructions?  What about storing all these snapshots?  This needs comments explaining whether it's just a simple example or is a model for how all tools should operate.

Reply:
> Are these snapshots supposed to be relative, or absolute?

The snapshot is absolute.

> Is there a convention to be documented? 

Not really. I think it's up to the tool to choose whether to track/print relative or absolute values, whatever is more useful to its users.

> What about storing all these snapshots?

Sure I can add a bit more sophisticated example.

Reply:
Done.